plugins/module_utils/redfish_utils.py

```diff
@@ -3983,3 +3983,7 @@ def get_power_restore_policy(self, systems_uri):
 
     def get_multi_power_restore_policy(self):
         return self.aggregate_systems(self.get_power_restore_policy)
+
+    def set_power_restore_policy(self, policy):
+        body = {'PowerRestorePolicy': policy}
```

Comment:
Can we add a check here that `policy` is within a set of enum values ("AlwaysOn", "AlwaysOff", or "LastState")?

Reply:
Hello,

Sure I can add a check here for a valid `PowerRestorePolicyTypes` enum option. But what is the rationale behind essentially duplicating it here and not depending on the Redfish backend to validate?

Reply:
Also using `choices` for the option might be the better way to check this. Then it's also part of the option's documentation.

Reply:
It's mainly to perform up-front validation the user's request is valid. Relying on the Redfish service to reject the request causes unnecessary traffic to the BMC,

Reply:
Please excuse me being very new to Ansible module development, I was basically following the already available code nearby as examples and missed the `choices` feature.
I hope now it's doing the right thing.
And yes, now I see how verifying the option validity prior to trying to send it to a hundred of servers makes sense, thank you for clarifying.**

Reply:
hi @paulfertser , we were all at that stage at some point, and we're happy to teach, just like we have been taught once. All good.